### PR TITLE
Update for wider graphgl use cases, and an update to the readme( you need GUT installed first) 

### PR DESCRIPTION
**Pull request by Mylab6**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
 # Godot GraphQLClient Addon
+
+## Install GUT 
+Before installing this, make sure to install GUT otherwise this addon won't work 
+See 
+https://gut.readthedocs.io/en/latest/index.html
+
```
Comment (Dracks) [inline README.md:7]: Quick question, what was the error without having gut installed?

File: .gitignore [hunk 2]
```diff
@@ -13,3 +13,4 @@ export_presets.cfg
 .mono/
 data_*/
 mono_crash.*.json
+**/*.uid
```
Comment (Dracks) [inline .gitignore]: are you sure this should be ignored? Keep in mind that the uid file was added into godot to make it more friendly with git/vcs

File: src/gql_query.gd [hunk 3]
```diff
@@ -27,18 +27,31 @@ func _serialize_args()->String:
...
-				query +=prop.serialize()+"\n"
-		query +="}"
+				query += prop.name
+				if prop.args_list.size() > 0:
+					query += " " + prop._serialize_args_usage()
+				query += " {\n"
+				for p in prop.props_list:
+					query += p + "\n"
+				query += "}\n"
+		query += "}"
```
Comment (Dracks) [inline src/gql_query.gd:56]: Can you create a test case for the edge case you are fixing? Because the original prop.serialize should be able to handle also the args_list (is a recursive function) and with the new code, I'm not sure it will be able to handle more than 2 layers of props/objects.

File: README.md (hunk 1 above)
Comment (Mylab6) [reply README.md:7]: Failed to load script "res://Addons/GodotGraphQL/tests/test_gql_query.gd" with error "Parse error".

File: .gitignore (hunk 2 above)
Comment (Mylab6) [reply .gitignore]: Agreed, will remove

File: src/gql_query.gd (hunk 3 above)
Comment (Mylab6) [reply src/gql_query.gd:56]: Let me double check this, and I'll get back to you.

File: .gitignore (hunk 2 above)
Comment (Mylab6) [reply .gitignore]: Removed.

File: src/gql_query.gd (hunk 3 above)
Comment (Mylab6) [reply src/gql_query.gd:56]: Here's the error I get with the original code 
```
Query:query getAllCards ($limit: Int) {
allCards (Int: $limit) {
name
type
effects{effectType}
}
}
Response:{"errors": [{"message": "Unknown argument 'Int' on field 'Query.allCards'.", "locations": [{"line": 2, "column": 11}], "extensions": {"exception": null}}]}
🔥 Got GraphQL response:
{ "errors": [{ "message": "Unknown argument \'Int\' on field \'Query.allCards\'.", "locations": [{ "line": 2.0, "column": 11.0 }], "extensions": { "exception": <null> } }] }
```
 I think the original code doesn't handle the limit int properly .
 
 My change supports nested objects , I'm able to get this result.( from a card game I'm working on, will open source soon once it's ready). 
 

This code ( when used against my branch).
```
extends GQLClient
@export var endpointSet: String

func _ready():
	endpoint = endpointSet 
	use_ssl = true
	var query = GQLQuery.new("allCards").set_args({"limit": "Int"}).set_props(["name","type","effects{effectType}"])
	var exec = query("getAllCards", {"limit": "Int"}, query)
	add_child(exec)
	exec.connect("graphql_response", Callable(self, "_on_graphql_response"))
	exec.run({"limit": 3})
func _on_graphql_response(data):
	print("🔥 Got GraphQL response:")
	print(data)
	var rich_text_node = $databox as RichTextLabel
	if rich_text_node:
		rich_text_node.clear()  # Clear previous text
		rich_text_node.add_text(JSON.stringify(data))
```

Produces :`
 { "data": { "allCards": [{ "name": "Data Cap", "type": "driver", "effects": [{ "effectType": "custom" }] }, { "name": "Shareware Demo", "type": "apiCall", "effects": [{ "effectType": "draw_next_turn" }] }, { "name": "DRAM Protection 07896", "type": "sysAgent", "effects": [{ "effectType": "custom" }] }] } }`

Comment (Dracks) [reply src/gql_query.gd:56]: I don't remember with details, but can you try with my original code something like:

```
func _ready():
	endpoint = endpointSet 
	use_ssl = true
	var query = GQLQuery.new("allCards").set_args({"limit": "limit"}).set_props(["name","type","effects{effectType}"])
	var exec = query("getAllCards", {"limit": "Int"}, query)
	add_child(exec)
	exec.connect("graphql_response", Callable(self, "_on_graphql_response"))
	exec.run({"limit": 3})
```

If I remember correctly when I did it I dessigned to work like that. 

Otherwise, this afternoon I will take a deeper look, since now I'm working. 

(I edited the code, that I modified it wrong, the query("getAllCards" was ok, the subquery was the wrong one)

Comment (Dracks) [reply src/gql_query.gd:56]: Here is an example of how I used to pass the variable into the function: 
https://github.com/Dracks/godot-gql-test/blob/main/main.gd#L9

Check if this docs help you:
https://github.com/Dracks/GodotGraphQL/pull/7

Also, I realise the way that GQLQuery accept the args right now can be confusing, as the key is the variable and the input parameter is the value. (Which can provocate problems, I will try to fix it now someway somehow)

Comment (Mylab6) [reply src/gql_query.gd:56]: Thank you,  this works with your original code, feel free to cancel this PR and add a note to the Readme to install GUT.

Comment (Dracks) [reply src/gql_query.gd:56]: I will try to do this afternoon, look at my other P/R, that I add a new method, that should make things easy, also it has a shortcut for when they are the same.

**Comment by Dracks** [comment]
Thanks for the P/R, as agreed I close it.
